Tasks/CopyFilesOverSSHV0/sshhelper.ts

```diff
@@ -103,31 +101,33 @@ export class SshHelper {
     async uploadFile(sourceFile: string, dest: string) : Promise<string> {
         tl.debug('Upload ' + sourceFile + ' to ' + dest + ' on remote machine.');
 
-        var defer = Q.defer<string>();
-        if(!this.sftpClient) {
-            defer.reject(tl.loc('ConnectionNotSetup'));
+        if (!this.sftpClient) {
+            return Promise.reject(tl.loc('ConnectionNotSetup'));
         }
 
-        const remotePath = path.dirname(dest);
-        try {
-            if (!await this.sftpClient.exists(remotePath)) {
-                await this.sftpClient.mkdir(remotePath, true);
-            }
-        } catch (error) {
-            defer.reject(tl.loc('TargetNotCreated', remotePath));
+        if (this.sshConfig.useFastPut) {
```

Comment:
missing old code to for exists(path)/mkdir

Reply:
Folders creation has been moved to a [stage](https://github.com/microsoft/azure-pipelines-tasks/pull/19798/files/845f325d885d5f6dc5f5df2ccac48bd9502af8c0#diff-00de2e31f64cd05be13d4503059e881bc13bbca5a61d1e75ef90757ab643d004R245) where the **contents** input is not empty.
Now folders creation is based on a result of **getUniqueFolders** function